app/src/main/java/org/mozilla/focus/ext/Fragment.kt

```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.ext
+
+import android.support.v4.app.Fragment
+
+val Fragment?.isVisibleAndNonNull: Boolean
+    get() = this != null && this.isVisible
```

Comment:
@Baron-Severin To make sure we're on the same page, what are your concerns with querying view state?

My understanding is that the complex logic for understanding what the current state is and updating it accordingly is 1) interspersed throughout several other complex files, 2) is not easily testable because of that, and 3) is encapsulated with other responsibilities (e.g. lifecycle), making it harder to follow.

@Baron-Severin  Do you have a suggestion about how I could improve this?

I originally was attempting to refactor this code to an MVVM style but it was taking too long. Thinking about it know, maybe I was trying to do too much at once and I could have started by moving the "which view(s) are displayed" into a model by itself.

Reply:
fwiw, potentially moving to MVVM in #167 so we can address this then.

Reply:
Those are definitely concerns, and there are a few more I can think of.  It ties our logic to our UI.  Say, for example, that we make a habit of checking whether or not we're on the `NavigationOverlayFragment` by checking its visibility.  Then we want to change settings to be a modal that appears on top of the overlay, which is still visible.  Suddenly a lot of our logic breaks in areas totally unrelated to the actual change, in a way that's not easy to fix.  

Another problem that comes to mind is that most fragment transactions are asynchronous, so we could be writing very narrow race conditions into our code by checking view state.  Animations are another factor that can cause problems (what if we want a view to animate out for 200ms before the new one becomes available, but the rest of the code expects it to be there immediately?)  These are all problems that are very hard to refactor out.

As for solutions, I'm not sure.  Navigation is tough and I don't know as much about it as I should.  I think a simple first step would be to check for fragment availability through an abstraction.  Even if `Navigator#isBrowserFragmentAvailable` queries view state under the hood, at least it gives us a good seam from which to change the implementation later.